fuse/eval/metrics/metrics_common.py

```diff
@@ -920,7 +941,11 @@ def eval(
                 stratum_filter = stratum_id == stratum
                 n_stratum = sum(stratum_filter)
                 random_sample = rnd.randint(0, n_stratum, size=n_stratum)
-                sampled_ids[stratum_filter] = ids[stratum_filter][random_sample]
+
+                flt_indx = np.where(stratum_filter)[0]
```

Comment:
You needed to move to "for loop implementation" even though you converted it ids to numpy array?